.github/workflows/reusable_tutorials.yml

```
@@ -90,12 +90,23 @@ jobs:
         python scripts/run_tutorials.py -p "$(pwd)"
     - if: ${{ inputs.upload_artifact }}
       name: Upload performance data to artifacts branch
-      # Upload any csv files (the tutorials script ensures there is only one)
+      # 1) Switch to artifacts branch
+      # 2) Add the new CSV file
+      # 3) Get .py and .ipynb files needed for analyzing tutorials data from main
+      # 4) Run `scripts/analyze_tutorials_performance.py` to merge CSVs and
+      #  update notebook
+      # 5) Commit updated notebook and new CSVs to artifacts branch
       run: |
         git config user.email "github-actions@github.com"
         git config user.name "github-actions"
+        cp scripts/analyze_tutorials_performance.py analyze_tutorials_performance.py
+        cp notebooks/tutorials_performance_tracking.ipynb tutorials_performance_tracking.ipynb
         git switch artifacts
+        mkdir scripts/
+        mv analyze_tutorials_performance.py scripts/
         mv *.csv tutorial_performance_data/
-        git add tutorial_performance_data/*.csv
-        git commit tutorial_performance_data/*.csv -m "Adding most recent tutorials output"
+        mv tutorials_performance_tracking.ipynb notebooks/
+        python scripts/analyze_tutorials_performance.py
```

We don't need to add Seaborn as a dependency if we're only using it here, although it might be nice to use it more widely.

```suggestion
        mv tutorials_performance_tracking.ipynb notebooks/
        pip install seaborn
        python scripts/analyze_tutorials_performance.py
```